tests/simulator/test_restarts.py

```diff
@@ -162,58 +213,87 @@ def test_restarts(self, dim, interp, simInput, expected_num_levels):
+            hi_cell = 19
+            b0 = [ [lo_cell]*dim, [hi_cell]*dim ]
+            simput["refinement_boxes"] = {"L0": {"B0": b0}}
+        else: # https://github.com/LLNL/SAMRAI/issues/199
+          # tagging can handle more than one timestep as it does not
+          #  appear subject to regridding issues, so we make more timesteps
+          #  to confirm simulations are still equivalent
+            simput["time_step_nbr"] = 10
 
-        n_quantities_per_patch = 20
-        pops = model.populations
-        for time in timestamps:
-            checks = 0
+        # if restart time exists it "loads" from restart file
+        #  otherwise just saves restart files based on timestamps
+        assert "restart_time" not in simput["restart_options"]
 
-            run0 = Run(diag_dir0)
-            run1 = Run(diag_dir1)
-            checks += check_particles(run0.GetParticles(time, pops), run1.GetParticles(time, pops))
-            checks += check_field(run0.GetB(time), run1.GetB(time))
-            checks += check_field(run0.GetE(time), run1.GetE(time))
-            checks += check_field(run0.GetNi(time), run1.GetNi(time))
-            checks += check_field(run0.GetVi(time), run1.GetVi(time))
+        simput["interp_order"] = interp
+        time_step = simput["time_step"]
+        time_step_nbr = simput["time_step_nbr"]
 
-            for pop in pops:
-                checks += check_field(run0.GetFlux(time, pop), run1.GetFlux(time, pop))
-                checks += check_field(run0.GetN(time, pop), run1.GetN(time, pop))
+        restart_idx = 4
+        restart_time=time_step * restart_idx
+        timestamps = [time_step * restart_idx, time_step * time_step_nbr]
 
-            n_levels, n_patches = count_levels_and_patches(run0.GetB(time))
-            self.assertEqual(n_levels, expected_num_levels) # at least 2 levels, 3 for tagging
-            self.assertGreaterEqual(n_patches, n_levels) # at least one patch per level
-            self.assertEqual(checks, n_quantities_per_patch * n_patches)
+        # first simulation
+        local_out = f"{out}/elapsed_test/{dim}/{interp}/mpi_n/{cpp.mpi_size()}/id{self.ddt_test_id()}"
+        simput["restart_options"]["dir"] = local_out
+        import datetime
+        seconds=9
+        simput["restart_options"]["elapsed_timestamps"] = [datetime.timedelta(seconds=seconds)]
+        simput["diag_options"]["options"]["dir"] = local_out
+        ph.global_vars.sim = None
+        ph.global_vars.sim = ph.Simulation(**simput)
+        self.assertEqual([seconds], ph.global_vars.sim.restart_options["elapsed_timestamps"])
+
+        assert "restart_time" not in ph.global_vars.sim.restart_options
+        model = setup_model()
+        dump_all_diags(model.populations, timestamps=np.array(timestamps))
+        simulator = Simulator(ph.global_vars.sim).initialize()
+        for i in range(restart_idx - 1):
+            simulator.advance()
+        import time
+        time.sleep(10)
```

Comment:
to avoid future mistakes I think it would probably be safer to write `time.sleep(seconds+1)`